exporters/trace/src/main/java/com/google/cloud/operations/opentelemetry/exporter/util/ResourceUtils.java

```diff
@@ -0,0 +1,76 @@
+package com.google.cloud.operations.opentelemetry.exporter.util;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import io.opencensus.resource.Resource;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public final class ResourceUtils {
+  static final Map<String, String>  EMPTY_RESOURCE_LABELS = Collections.<String, String>emptyMap();
+
+  /**
+   * Returns a set of labels for detected resource.
+   * Detector sequentially runs resource detection from environment
+   * variables, K8S, GCE and AWS.
+   *
+   * @return a set of resource labels.
+   */
+  public static Map<String, String> detectResourceLabels() {
```

Comment:
Not sure if this code should be ported from OpenCensus.
We don't have similar auto-detection in Go AFAIK, and auto labels could actually be implemented in a generic way as part of OpenTelemetry SDK rather than specific to Cloud Trace.

Reply:
Removed for now.